src/scripts/main.js

```diff
@@ -1,3 +1,320 @@
 'use strict';
 
-// write code here
+const tableHead = document.querySelector('thead');
+const tableBody = document.querySelector('tbody');
+const tableRows = tableBody.rows;
+let toggle = true;
+let valueForSort = 0;
+let special;
+
+function setSalaryValue(num) {
+  return `$${num.toLocaleString('en-US')}`;
+};
+
+function getSalaryValue(str) {
+  return +str.replace(/[,$]/g, '');
+};
+
+tableHead.addEventListener('click', e => {
+  const index = e.target.cellIndex;
+  const headTitle = e.target.innerHTML;
+
+  const sortRows = [...tableRows].sort((a, b) => {
+    let prev = a.cells[index].innerHTML;
+    let next = b.cells[index].innerHTML;
+
+    if (headTitle === 'Salary' || headTitle === 'Age') {
+      prev = getSalaryValue(prev);
+      next = getSalaryValue(next);
+
+      return prev - next;
+    }
+
+    return prev.localeCompare(next);
+  });
+
+  if (!toggle) {
+    sortRows.reverse();
+  }
+
+  tableBody.append(...sortRows);
+
+  if (valueForSort !== index) {
+    toggle = !toggle;
+  }
+
+  toggle = !toggle;
+
+  valueForSort = index;
+});
+
+tableBody.addEventListener('click', (e) => {
+  if (special) {
+    special.classList.remove('active');
+  }
+
+  special = e.target.parentElement;
+  e.target.parentElement.classList.add('active');
+});
+
+document.body.insertAdjacentHTML('beforeend', `
+  <form
+    action="/"
+    method="get"
+    class="new-employee-form"
+  >
+    <label>
+      Name:
+      <input
+        name="name"
+        type="text"
+        data-qa="name"
+        minlength="4"
+        required
+      >
+    </label>
+    <label>
+      Position:
+      <input
+        name="position"
+        type="text"
+        data-qa="position"
+        required
+      >
+    </label>
+    <label for="office">
+      Office:
+      <select
+        id="office"
+        data-qa="office"
+        required
+      >
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age:
+      <input
+        name="age"
+        type="number"
+        data-qa="age"
+        required
+        min="18"
+        max="90"
+      >
+    </label>
+    <label>
+      Salary:
+      <input
+        name="salary"
+        type="number"
+        min = "0"
+        data-qa="salary"
+        required
+      >
+    </label>
+    <button type="submit"> Save to table </button>
+  </form>
+`);
+
+const form = document.forms[0];
+const officeList = form.querySelector('[data-qa="office"]');
+
+function createRow() {
```

Comment:
Consider renaming the 'createRow' function to 'addRowToTable' for better readability.